src/tsickle.ts

```diff
@@ -765,51 +765,68 @@ class Annotator extends ClosureRewriter {
-        // @fileoverview, @modName etc comments.
-        this.skipUpToOffset = comments[0].end;
-      }
+    this.emitSuppressChecktypes(sf);
+    for (const stmt of sf.statements) {
+      this.visit(stmt);
     }
-    if (fileoverviewTags) {
-      // Add @suppress {checkTypes}, or add to the list in an existing @suppress tag.
-      // Closure compiler barfs if there's a duplicated @suppress tag in a file, so the tag must
-      // only appear once and be merged.
-      const suppressIdx = fileoverviewTags.findIndex(t => t.tagName === 'suppress');
-      if (suppressIdx !== -1) {
-        const suppressions =
-            (fileoverviewTags[suppressIdx].text || '').replace(/^\{(.*)\}$/, '$1').split(',');
-        if (suppressions.indexOf('checkTypes') === -1) {
-          suppressions.push('checkTypes');
-        }
-        fileoverviewTags[suppressIdx].text = `{${suppressions.join(',')}}`;
-      } else {
-        fileoverviewTags.push({tagName: 'suppress', text: '{checkTypes}'});
+    if (sf.statements.length) {
+      this.writeRange(sf.statements[sf.statements.length - 1].getEnd(), sf.getEnd());
+    }
+  }
+
+  /**
+   * Emits an \@suppress {checkTypes} fileoverview comment.
+   */
+  private emitSuppressChecktypes(sf: ts.SourceFile) {
+    const comments = ts.getLeadingCommentRanges(sf.getFullText(), 0) || [];
+    let fileoverviewIdx = -1;
+    for (let i = comments.length - 1; i >= 0; i--) {
+      const parsed = jsdoc.parse(sf.getFullText().substring(comments[i].pos, comments[i].end));
+      if (parsed !== null && parsed.tags.some(t => t.tagName === 'fileoverview')) {
+        fileoverviewIdx = i;
+        break;
       }
-    } else {
-      fileoverviewTags = [
+    }
+    // Add a @suppress {checkTypes} tag to each source file's JSDoc comment,
+    // being careful to retain existing comments and their @suppress'ions.
+    // Closure Compiler considers the *last* comment with @fileoverview (or @externs or @nocompile)
+    // that has not been attached to some other tree node to be the file overview comment, and
+    // only applies @suppress tags from it.
+    // AJD considers *any* comment mentioning @fileoverview.
+    if (fileoverviewIdx === -1) {
+      // No existing comment to merge with, just emit a new one.
+      this.emit(jsdoc.toString([
         {tagName: 'fileoverview', text: 'added by tsickle'},
         {tagName: 'suppress', text: '{checkTypes}'}
-      ];
+      ]));
+      this.emit('\n');
+      return;
     }
-    this.emit(jsdoc.toString(fileoverviewTags));
-    this.emit('\n\n');  // separate from file body to avoid being dropped.
-    for (const stmt of sf.statements) {
-      this.visit(stmt);
+    const comment = comments[fileoverviewIdx];
+    this.writeRange(0, comment.pos);
+    this.skipUpToOffset = comment.end;
+
+    const parsed = jsdoc.parse(sf.getFullText().substring(comment.pos, comment.end));
+    if (!parsed) throw new Error('parsed just before?');
```

Comment:
This is a "this cannot happen" case, the code above has just validate that the thing at the index can be parsed. I've made it a bit more clear this is an internal error.